app/controllers/compute_resources_controller.rb

```diff
@@ -42,19 +42,24 @@ def edit
   end
 
   def associate
-    count = 0
-    if @compute_resource.respond_to?(:associated_host)
-      @compute_resource.vms(:eager_loading => true).each do |vm|
-        if Host.for_vm(@compute_resource, vm).empty?
-          host = @compute_resource.associated_host(vm)
-          if host.present?
-            host.associate!(@compute_resource, vm)
-            count += 1
-          end
-        end
+    if @compute_resource.supports_host_association?
+      associator = ComputeResourceHostAssociator.new(@compute_resource)
+      associator.associate_hosts
+      messages = []
+      if associator.hosts.empty?
+        messages << _('No VMs matched any host.')
+      else
+        messages << n_('%s VM was associated to a host.', '%s VMs were each associated to hosts.', associator.hosts.count) % associator.hosts.count
+      end
+      if associator.fail_count > 0
+        messages << _('%s VMs failed while processing: check logs for more details.') % associator.fail_count
+        process_error(:error_msg => messages.join(', '))
```

Comment:
Now, that the messages have a dot, they cannot be joined by ', '. This will lead to '., '.

Reply:
Updated